test/functional/tool_wallet.py

```diff
@@ -207,10 +215,25 @@ def test_salvage(self):
         # TODO: Check salvage actually salvages and doesn't break things. https://github.com/bitcoin/bitcoin/issues/7463
         self.log.info('Check salvage')
         self.start_node(0, ['-wallet=salvage'])
+        address_gen_rand = random.Random(123)
+
+        def gna():
+            return self.nodes[0].getnewaddress(
+                address_type=address_gen_rand.choice(['legacy', 'p2sh-segwit', 'bech32']))
+
+        for _ in range(101):
+            # Generate to fresh address each time
+            self.nodes[0].generatetoaddress(1, gna())
+        self.nodes[0].sendmany(amounts={gna(): 0.05 for _ in range(50)})
+        self.nodes[0].sendmany(amounts={gna(): 0.05 for _ in range(50)})
+        balances_before = self.nodes[0].getbalances()
         self.stop_node(0)
 
         self.assert_tool_output('', '-wallet=salvage', 'salvage')
 
+        self.start_node(0, ['-wallet=salvage'])
```

Comment:
I had to add `self.nodes[0].rescanblockchain(0)` to make this pass after merging with #19457